test/nodetool/test_snapshot.py

```diff
@@ -99,3 +102,201 @@ def test_listsnapshots(nodetool, request):
         assert res == scylla_expected_output
     else:
         assert res == cassandra_expected_output
+
+
+def check_snapshot_out(res, tag, ktlist, skip_flush):
+    """Check that the output of nodetool snapshot contains the expected messages"""
+
+    if len(ktlist) == 0:
+        keyspaces = "all keyspaces"
+    else:
+        keyspaces = ", ?".join(ktlist)
+
+    pattern = re.compile("Requested creating snapshot\\(s\\)"
+                         f" for \\[{keyspaces}\\]"
+                         f" with snapshot name \\[(.+)\\]"
+                         f" and options \\{{skipFlush={str(skip_flush).lower()}\\}}")
+
+    print(res)
+
+    lines = res.split("\n")
+    assert len(lines) == 3
+
+    match = pattern.fullmatch(lines[0])
+    assert match
+    actual_tag = match.group(1)
+    if type(tag) is int:
+        # The tag is a timestamp in millis.
+        # Allow for 10s worth of divergence between actual and expected tag value.
+        assert int(actual_tag) - tag <= 10000
```

Comment:
You don't have here abs(), so if the difference is negative, it can be very very negative and it's fine. Is this intentional?

Reply:
By the way, given the crazy-slow-tests we see once in a while, is 10 seconds enough? :-)

Reply:
In practice `actual` will always be larger than `tag`, because `tag` is generated by the test and `actual_tag` is generated by nodetool, at a later point in time. But just to be on the safe size, I will increase this.
I will also increase the allowed difference to 30sec.

Reply:
Yes, in practice this test will always pass because nodetool really does what we expect it to do ;-) But isn't the point to check that it does?